src/matter/cluster/LevelControlCluster.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OptionalAttribute, WritableAttribute, OptionalWritableAttribute, Attribute, Cluster, Command, OptionalCommand, TlvNoResponse } from "./Cluster";
+import { BitFlag, MatterApplicationClusterSpecificationV1_0, TlvBitmap, TlvEnum, TlvField, TlvNullable, TlvObject, TlvUInt16, TlvUInt8 } from "@project-chip/matter.js";
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.5.8 */
+const optionsBitmap = TlvBitmap(TlvUInt8, {
+    /** Dependency on On/Off cluster. */
+    executeIfOff: BitFlag(0), // TODO default: true.
+
+    /** Level effect on color. */
+    coupleColorTempToLevel: BitFlag(1),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.1 */
+const MoveToLevelCommandRequest = TlvObject({
+    level: TlvField(0, TlvUInt8.bound({max:254})),
+    transitionTime: TlvField(1, TlvNullable(TlvUInt16)),
+    optionsMask: TlvField(2, optionsBitmap),
+    optionsOverride: TlvField(3, optionsBitmap), // TODO: 0 Default - all optionsOverride
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.2 */
+const MoveCommandRequest = TlvObject({
+    moveMode: TlvField(0, TlvEnum<MoveMode>()),
+    rate: TlvField(1, TlvNullable(TlvUInt8)),
+    optionsMask: TlvField(2, optionsBitmap),
+    optionsOverride: TlvField(3, optionsBitmap),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.3 */
+const StepCommandRequest = TlvObject({
+    stepMode: TlvField(0, TlvEnum<StepMode>()),
+    stepSize: TlvField(1, TlvUInt8),
+    transitionTime: TlvField(2, TlvNullable(TlvUInt16)),
+    optionsMask: TlvField(3, optionsBitmap),
+    optionsOverride: TlvField(4, optionsBitmap),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.4 */
+const StopCommandRequest = TlvObject({
+    optionsMask: TlvField(0, optionsBitmap),
+    optionsOverride: TlvField(1, optionsBitmap),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.5  */
+const MoveToClosestFrequencyCommandRequest = TlvObject({
+    frequency: TlvField(0, TlvUInt16), // TODO - Default 0, how to set TlvField default?
+})
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.4 */
+const features = {
+    /** Dependency with the On/Off cluster */
+    onOffClusterDependency: BitFlag(0), // TODO default: true.
+
+    /** Behavior that supports lighting applications. */
+    lighting: BitFlag(1),
+
+    /** Supports frequency attributes and behavior. */
+    supportFrequency: BitFlag(2)
+};
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.6.2.1 */
+const enum MoveMode {
+    Up = 0x0,
+    Down = 0x1,
+};
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.6.3 */
+const enum StepMode {
+    Up = 0x0,
+    Down = 0x1,
+};
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.5 */
+const attributes = {
```

Comment:
Maybwe rename to commonAttributes and have all in that are not especially LT or FQ and add them then separately ? (see below example for commands)

(Sorry for all these cpmments .. in fact isit is one of the first bigger clusters, so we learn how we want to do it in the future )

Reply:
I _think_ I accomplished this and the "commands" comment by breaking out the frequency specific commands and attributes and then using the "Merge()" function to add them back in before the definition of the "PulseWidthModulationLevelControlCluster". But then again, I'm a Typescript novice so when I push, please evaluate what I did - perhaps there's a better way?

No problem on the large number of comments. This is a learning experience for me and I appreciate the feedback.

Reply:
I think "merge2 is not needed ... just do it as I showed in https://github.com/mfucci/node-matter/pull/255#discussion_r1117056833

Reply:
I had trouble trying to incorporate as you commented. Just pushed what I have. Probably still needs work